StorageUtility/Windows/Device.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+<License>
+Copyright 2015 Virtium Technology
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+</License>
+*/
+#include "Device.h"
+#include "StorageUtility.h"
+
+namespace vtStor
+{
+
+cDevice::cDevice(const PSP_DEVINFO_DATA& DevInfoData, const PSP_DEVICE_INTERFACE_DATA& DevInterfaceData, const PSP_INTERFACE_DEVICE_DETAIL_DATA& DevDetailData)
+{
+    U32 size = sizeof(DevInfoData->cbSize) + sizeof(DevInfoData->ClassGuid) + sizeof(DevInfoData->DevInst) + sizeof(DevInfoData->Reserved);
+    m_DevInfoData = (PSP_DEVINFO_DATA)LocalAlloc(LPTR, size);
+    m_DevInfoData->cbSize = DevInfoData->cbSize;
+    m_DevInfoData->ClassGuid = DevInfoData->ClassGuid;
+    m_DevInfoData->DevInst = DevInfoData->DevInst;
+    m_DevInfoData->Reserved = DevInfoData->Reserved;
+
+    size = sizeof(DevInterfaceData->cbSize) + sizeof(DevInterfaceData->Flags) + sizeof(DevInterfaceData->InterfaceClassGuid) + sizeof(DevInterfaceData->Reserved);
+    m_DevInterfaceData = (PSP_DEVICE_INTERFACE_DATA)LocalAlloc(LPTR, size);
+    m_DevInterfaceData->cbSize = DevInterfaceData->cbSize;
+    m_DevInterfaceData->Flags = DevInterfaceData->Flags;
+    m_DevInterfaceData->InterfaceClassGuid = DevInterfaceData->InterfaceClassGuid;
+    m_DevInterfaceData->Reserved = DevInterfaceData->Reserved;
+
+    U32 devicePathLength = (std::wcslen(DevDetailData->DevicePath) + 1);
+    size = sizeof(DevDetailData->cbSize) + (devicePathLength * sizeof(DevDetailData->DevicePath));
+    m_DevDetailData = (PSP_INTERFACE_DEVICE_DETAIL_DATA)LocalAlloc(LPTR, size);
+    m_DevDetailData->cbSize = DevDetailData->cbSize;
+    wmemcpy(m_DevDetailData->DevicePath, DevDetailData->DevicePath, devicePathLength);
```

Comment:
1. This can be simplify to two functions call: (1) allocate memory, and (2) copy memory.
2. LocalAlloc and LocalFree should be replaced with heap functions. This was discuss in the first meeting (see email issue #5).
3. The size of this structure has already been determine, so just pass it in as a parameter.